inference-engine/src/mkldnn_plugin/nodes/mkldnn_embedding_bag_sum_node.cpp

```diff
@@ -35,10 +35,30 @@ MKLDNNEmbeddingBagSumNode::MKLDNNEmbeddingBagSumNode(
              IE_THROW() << logPrefix << "must have equal shapes for indices and per_sample_weights inputs.";
     }
 
-    const auto& inDataDims = op->get_input_shape(EMB_TABLE_IDX);
+    if (!isDynamicNgraphNode(op)) {
+        const auto& inDataDims = op->get_input_shape(EMB_TABLE_IDX);
+        _embDepth = 1lu;
+        for (size_t i = 1lu; i < inDataDims.size(); i++) {
+            _embDepth *= inDataDims[i];
+        }
+    }
+}
+
+const VectorDims& MKLDNNEmbeddingBagSumNode::getStaticShape(const MKLDNNEdgeWeakPtr& parentEdgePtr) {
+    const auto parentEdge = parentEdgePtr.lock();
+    assert(parentEdge != nullptr &&  "parent edge is nullable");
+
+    const auto& srcMemory = parentEdge->getMemoryPtr();
+    assert(srcMemory != nullptr &&  "parent edge memory is nullable");
+
+    return srcMemory->getStaticDims();
+}
+
+void MKLDNNEmbeddingBagSumNode::prepareParams(const std::vector<MKLDNNEdgeWeakPtr>& parentEdges) {
+    const auto& staticShape = getStaticShape(parentEdges[EMB_TABLE_IDX]);
```

Comment:
I guess we shouldn't use such a complex way.
Better to remove `MKLDNNEmbeddingBagSumNode::getStaticShape` at all, make `VectorDims` parameter for `MKLDNNEmbeddingBagSumNode::prepareParams`. And call inside all `MKLDNNEmbedding*::prepareParams` `MKLDNNEmbeddingBagSumNode::prepareParams(getParentEdgesAtPort(EMB_TABLE_IDX)[0]->getMemory().getStaticDims()`

Reply:
done, thanks